### PR TITLE
feat(debug-metadata): expose customizable sourceMappingURL rewriter

### PR DESCRIPTION
**Pull request by upupming**
## Summary

Expose `rewriteSourceMappingURLs` and `RewriteSourceMappingURLsOptions` from `@lynx-js/debug-metadata-rsbuild-plugin`. Consumers can call the rewriter from their own `beforeEncode` tap and supply a `getSourceMappingURL` callback to customize the per-asset URL — e.g. point each chunk's directive at a gateway that supports server-side field filtering.

## Motivation

The plugin already rewrites `//# sourceMappingURL=...` directives to an absolute URL in dev (when `devServerOrigin` is provided), using `${debugMetadataUrl}?field=source-map&path=<encoded mapPath>` — a container pattern that downloads the whole metadata client-side.

Downstream plugins that compute a content hash at build time and upload the metadata to a gateway with field filtering need to bake a per-chunk URL of the form `?tos_key=<hash>&field=source-map&path=<mapPath>` instead, so consumers fetch a small per-chunk source map rather than the full container. Previously the rewriter was a private helper, leaving no way to share the iteration/regex without duplicating it.

## What changed

- `source-mapping-url-rewriter.ts`: factor out `rewriteSourceMappingURL(source, newUrl)` as the directive-replacement primitive. `rewriteSourceMappingURLToAbsolute` (formerly `rewriteTrailerToAbsoluteUrl`) becomes a thin wrapper that builds the default container URL and delegates.
- `LynxDebugMetadataPlugin.ts`: `rewriteSourceMappingURLs` (formerly file-private `rewriteSourceMappingURLTrailers`) is now exported. New optional `options.getSourceMappingURL: ({ mapPath, debugMetadataUrl }) => string` lets consumers compute a custom URL per asset. Default behavior unchanged.
- `index.ts`: re-exports `rewriteSourceMappingURLs` and `RewriteSourceMappingURLsOptions`.
- Unit tests updated to the renamed identifiers.

## Test plan

- [x] OSS package vitest: 91/91 pass.
- [x] tsc \`--build\` resolves the new exports.
- [x] Pre-commit (eslint / biome / dprint / cspell) passes.
- [x] api-extractor report (\`etc/debug-metadata-rsbuild-plugin.api.md\`) regenerated.

A changeset (\`minor\`) is included.

## Summary by CodeRabbit

* **New Features**
  * Added public rewrite capability to customize generated sourceMappingURL values via an optional callback.

* **Bug Fixes / Behavior**
  * Rewriter safely skips data: URLs and existing debug-metadata links; preserves comment style and rewrites only appropriate trailers.

* **Tests**
  * Expanded tests to validate JS/CSS trailer handling, idempotency, encoding, and edge cases.

* **Chores**
  * Release metadata added for a minor version.

### CODE REVIEW

File: packages/rspeedy/plugin-debug-metadata/src/LynxDebugMetadataPlugin.ts [hunk 1]
```diff
@@ -354,11 +376,14 @@ function rewriteSourceMappingURLTrailers(
...
     const before = asset.source.source().toString()
-    const after = rewriteTrailerToAbsoluteUrl(
-      before,
+    const mapPath = `${assetName}.map`
+    const customUrl = options?.getSourceMappingURL?.({
+      mapPath,
       debugMetadataUrl,
-      `${assetName}.map`,
-    )
+    })
```
[the remark this answers is not in the document]
Comment (upupming) [reply packages/rspeedy/plugin-debug-metadata/src/LynxDebugMetadataPlugin.ts:383]: Thanks — the caller is expected to set up `debugMetadataUrl` before invoking this helper even when supplying a custom `getSourceMappingURL`, so the early-return on an empty `debugMetadataUrl` is the intended safety: it gates the rewriter to builds that participate in the metadata flow. If a callback-only mode without `debugMetadataUrl` becomes a real use case, the right move is to promote the low-level `rewriteSourceMappingURL(source, newUrl)` (currently `@internal`) to `@public` so consumers iterate assets themselves — keeping the high-level helper's contract simple. Resolving as a no-op for this PR.

File: packages/rspeedy/plugin-debug-metadata/src/LynxDebugMetadataPlugin.ts [hunk 2]
```diff
@@ -354,11 +376,14 @@ function rewriteSourceMappingURLTrailers(
...
+    const after = customUrl === undefined
+      ? rewriteSourceMappingURLToAbsolute(before, debugMetadataUrl, mapPath)
+      : rewriteSourceMappingURL(before, customUrl)
```
[the remark this answers is not in the document]
Comment (upupming) [reply packages/rspeedy/plugin-debug-metadata/src/LynxDebugMetadataPlugin.ts:386]: Good catch — fixed in the latest commit. The idempotency guard has been moved from the low-level `rewriteSourceMappingURL` primitive into `rewriteSourceMappingURLToAbsolute` (default-flow only), so a consumer-supplied `getSourceMappingURL` now always overwrites — including in the dev case where the built-in tap already baked the container URL into the directive first. The low-level test was updated to assert overwrite-on-existing-container.

File: packages/rspeedy/plugin-debug-metadata/src/LynxDebugMetadataPlugin.ts [hunk 3]
```diff
@@ -233,7 +236,7 @@ export class LynxDebugMetadataPluginImpl {
               }`
           }
 
-          rewriteSourceMappingURLTrailers(compilation, args)
+          rewriteSourceMappingURLs(compilation, args)
```
[the remark this answers is not in the document]
Comment (upupming) [reply packages/rspeedy/plugin-debug-metadata/src/LynxDebugMetadataPlugin.ts:239]: By design — when a consumer plugin needs its `getSourceMappingURL` URL to be the final one baked into the directive, it should register its rsbuild plugin **after** `pluginLynxDebugMetadata` so its `beforeEncode` tap runs last. The built-in pass uses `rewriteSourceMappingURLToAbsolute`, which only skips already-rewritten `debug-metadata.json` directives; a custom gateway URL set by an earlier consumer pass would indeed be overwritten by the default. Trying to broaden the default's idempotency guard (e.g. skip any absolute http URL) would over-skip legitimate webpack absolute paths, so the tap-order contract is left to the consumer — and the helper docstring covers this in the “Call from a beforeEncode tap, after …” line. Resolving.

**Review by luhc228 — APPROVED** [review]
